libraries/cmake/formula/openssl/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.14.5)
+project(thirdparty_openssl)
+
+include(ExternalProject)
+
+function(opensslMain)
+  generateMetadataTargets()
+
+  if(NOT DEFINED thirdparty_zlib_INCLUDE_DIRS OR
+     NOT DEFINED thirdparty_zlib_LIBRARIES)
+
+    message(WARNING "The zlib dependency was not specified correctly")
+    return()
+  endif()
+
+  # The formula will be configured with the C_FLAGS and CXX_FLAGS parameters; we should
+  # forward them to the configure script when possible
+  message(WARNING "TODO: Use C_FLAGS")
+
+  # TODO(alessandro): Port all the settings from the original formula
+  # TODO(alessandro): Use the compilation flags we got from the main project (C_FLAGS, CXX_FLAGS)
+  ExternalProject_Add(
+    openssl
+    URL "https://www.openssl.org/source/openssl-1.0.2o.tar.gz"
+
+    CONFIGURE_COMMAND
+      "${CMAKE_COMMAND}" -E env CC="${CMAKE_C_COMPILER}"
+      perl ./Configure "--prefix=${CMAKE_INSTALL_PREFIX}" "--openssldir=${CMAKE_INSTALL_PREFIX}/etc/openssl"
+        no-ssl2
+        no-ssl3
+        no-asm
+        no-shared
+        no-weak-ssl-ciphers
+        zlib-dynamic
+        enable-cms
+        "--with-zlib-include=${thirdparty_zlib_INCLUDE_DIRS}"
+        "--with-zlib-lib=${thirdparty_zlib_LIBRARIES}"
+        linux-x86_64
+
+    BUILD_COMMAND
+     "${CMAKE_COMMAND}" -E env CC="${CMAKE_C_COMPILER}"
+     "${CMAKE_COMMAND}" -E make_directory "${CMAKE_INSTALL_PREFIX}/etc/openssl" &&
+      $(MAKE) depend -j 10 &&
```

Comment:
This is a totally arbitrary value that we should make configurable and default to a sane value

Reply:
You should be able to remove `-j 10` right now with the use of `$(MAKE)`.

Reply:
Does this use of `$(MAKE)` within the `BUILD_COMMAND` property work with Ninja as the generator?

Reply:
The formula system was implemented using custom commands that are triggered when the required library files are referenced. The build command is executed in a new environment and is detached from the main project. It should work in theory, but I didn't test it yet.

Reply:
I had to change to `make` because `$(MAKE)` is only valid if the generator calling the command is make; that's what you use for coordinating with the jobserver and avoid to spawn too many jobs.
It will spawn more jobs than you required when building osquery, but it shouldn't be too much of an issue, especially for building openssl.